### PR TITLE
Report memory limit and allocated memory in longrepr

### PR DESCRIPTION
**Pull request by petr-tik**
Make the longrepr more descriptive and enrich the xml output with
this information. 

Refactor limit_memory to return a `_MemoryInfo` type, which
can be presented as a PytestSection and a longrepr string.

Remove item.nodeid from longrepr since longrepr is only used by
junit-xml parser and the structure of the xml document
makes the item.nodeid in the error message redundant.

**Testing performed**
Add a test that parses the resulting xml file to make sure
the expected string was written.

*Issue number of the reported bug or feature request: #3*

### CODE REVIEW

**Comment by gaborbernat** [comment]
@petr-tik can you try to resolve the merge conflicts?

**Comment by petr-tik** [comment]
@gaborbernat i have. please take a look when you can

File: tests/test_pytest_pensieve.py
Comment (gaborbernat) [inline tests/test_pytest_pensieve.py:127]: save the string as a variable, also single quote is enough no need for triple

File: tests/test_pytest_pensieve.py
Comment (gaborbernat) [inline tests/test_pytest_pensieve.py:117]: One level of indentation less as other tests do

File: tests/test_pytest_pensieve.py
Comment (gaborbernat) [inline tests/test_pytest_pensieve.py:106]: type annotate all arguments

File: src/pytest_memray/marks.py
Comment (gaborbernat) [inline src/pytest_memray/marks.py:59]: explicit `return None`

File: src/pytest_memray/marks.py
Comment (gaborbernat) [inline src/pytest_memray/marks.py:23]: use new style annotations - List -> list

File: src/pytest_memray/marks.py
Comment (gaborbernat) [inline src/pytest_memray/marks.py:32]: this should be in one line array declration

File: src/pytest_memray/marks.py
Comment (gaborbernat) [inline src/pytest_memray/marks.py:41]: no need for surrounding (); keep the style present on main

File: src/pytest_memray/marks.py
Comment (gaborbernat) [inline src/pytest_memray/marks.py:25]: should this be just a section property instead?

File: src/pytest_memray/marks.py
Comment (gaborbernat) [inline src/pytest_memray/marks.py:43]: should this be just a long_repr property instead?

**Review by gaborbernat — CHANGES_REQUESTED** [review]

**Comment by petr-tik** [comment]
Thanks for the detailed review with code style and refactor suggestions.

Addressed all of them. Only have 1 REVIEW comment about the name of the dataclass, happy to remove the comment, if you find the type name descriptive enough.

File: src/pytest_memray/marks.py
Comment (gaborbernat) [inline src/pytest_memray/marks.py:17]: Let's remove the comment, I think we can keep it just `_MemoryInfo` (can be private, outside people don't need to use it) 👍

File: src/pytest_memray/marks.py
Comment (gaborbernat) [inline src/pytest_memray/marks.py:49]: Remove triple quote, single suffices; also can we shorten the variable names to fit this return into a single line?

File: tests/test_pytest_pensieve.py
Comment (gaborbernat) [inline tests/test_pytest_pensieve.py:113]: @pablogsal can we expose this publicly instead of doing this private import?

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: tests/test_pytest_pensieve.py
Comment (pablogsal) [reply tests/test_pytest_pensieve.py:113]: Nop, this is (and will remain as) a testing utility. I consider it the same as doing this import in memray itself.

Comment (gaborbernat) [reply tests/test_pytest_pensieve.py:113]: 👍

**Review by gaborbernat — APPROVED** [review]

**Comment by petr-tik** [comment]
do you want me to edit the PR message before merging?

**Comment by petr-tik** [comment]
i wanted to tidy it up and leave a descriptive commit message in main

**Comment by gaborbernat** [comment]
Go ahead with that then 👍

**Comment by petr-tik** [comment]
Done. I am guessing you are the one with squash and merge rights?

**Comment by pablogsal** [comment]
Thanks a lot @petr-tik for the fantastic work and for persuing getting this fixed ❤️ 

And thanks a lot @gaborbernat for the review! 💪

**Comment by petr-tik** [comment]
oh, my PR message wasn't included in the commit. 

Thanks for the detailed review. Absolute pleasure

**Pull request merged**
